OZON.ru.js

```diff
@@ -9,596 +9,130 @@
 
-	This program is distributed in the hope that it will be useful,
-	but WITHOUT ANY WARRANTY; without even the implied warranty of
-	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-	Affero General Public License for more details.
+	// translator, editor, etc.
+	var additionalProperties = iteratorToArray(
+		xpath('.//p[not(@itemprop)]', content)
+	);
+	for (i in additionalProperties) {
+		var property = getAdditionalProperty(additionalProperties[i]);
+		if (typeof property === 'undefined')
+			continue;
 
-	You should have received a copy of the GNU Affero General Public
-	License along with this program.  If not, see
-	<http://www.gnu.org/licenses/>.
-*/
+		if (property.name === 'Переводчик')
+			book.creators.push(createCreator('translate', property.value));
 
-function detectWeb(doc, url) {
-	var itemTypeXPath = '//ul[@class="navLine"]/li[1]/a';
-	var itemTypeDOMNode =
-	doc.evaluate(itemTypeXPath, doc, null,
-	XPathResult.ANY_TYPE, null).iterateNext();
-	var itemType;
-	if (itemTypeDOMNode) itemType = itemTypeDOMNode.textContent;
-	else return "";
-	if (itemType == "Книги") {
-		var nameXPath = '//div[@class="techDescription"]/div/div[2]/span';
-		var valueXPath = '//div[@class="techDescription"]/div/div[3]/span';
-		var nameXPathRes = doc.evaluate(
-		nameXPath, doc, null, XPathResult.ANY_TYPE, null);
-		var valueXPathRes = doc.evaluate(
-		valueXPath, doc, null, XPathResult.ANY_TYPE, null);
-		if (!IsAudioBook(nameXPathRes, valueXPathRes)) return "book";
-	} else {
-		itemTypeXPath = '//ul[@class="navLine"]/li[2]/a';
-		itemTypeDOMNode =
-		doc.evaluate(itemTypeXPath, doc, null,
-		XPathResult.ANY_TYPE, null).iterateNext();
-		if (itemTypeDOMNode) itemType = itemTypeDOMNode.textContent;
-		else return "";
-		if (itemType == "Цифровые книги") return "book";
-	}
-}
-
-function IsAudioBook(nameXPathRes, valueXPathRes) {
-	var nameDOMNode;
-	var valueDOMNode;
-	while( (nameDOMNode = nameXPathRes.iterateNext()) &&
-	(valueDOMNode = valueXPathRes.iterateNext()) ) {
-		var nameText = CleanText(nameDOMNode.textContent);
-		var valueText = CleanText(valueDOMNode.textContent);
-		if (nameText == "Формат аудиокниги") return true;
-	}
-	return false;
-}
+		if (property.name === 'Редактор')
+			book.creators.push(createCreator('editor', property.value));
 
-function MergeObjects(obj1, obj2) {
-	var result = new Object();
-	for (var attrName in obj1) {
-		result[attrName] = obj1[attrName];
+		if (property.name === 'Серия')
+			book.series = property.value;
 	}
-	for (var attrName in obj2) {
-		result[attrName] = obj2[attrName];
-	}
-	return result;
-}
 
-function CleanText(text) {
-	var result = text.replace(/^\s*|\s*$/g, '');
-	result = result.replace(/\s+/g, ' ');
-	return result;
+	book.complete();
 }
 
-function CleanCreator(creatorStr) {
-	return CleanText(creatorStr);
+function xpath(expr, context) {
+	// executes standard XPath search on a document
+    return document.evaluate(expr, context, null, XPathResult.ANY_TYPE, null);
```

Comment:
never refer to `document` directly. Use `doc` that is passed into `doWeb`, `detectWeb`, callback to `ZU.processDocuments`